### ISSUE TITLE
Upgrade to GraalVM 19.3.0 JDK11

### ISSUE THREAD

**Issue by borkdude** [borkdude/babashka#120]
See the JDK 11 branch.

- [x] The class java.lang.UNIXProcess no longer exists in JDK 11.
That functionality is now in java.lang.ProcessImpl.
So if you really need a reflective access to it, you need separate reflection configuration files for JDK 8 and JDK 11.

For now I removed the classes from the config, let's just see what happens.

- [x] Using clojure.lang.Reflector results in this error:
https://gist.github.com/borkdude/8ec406e82f0db128d5914e58563e44fc#file-graalvm-jdk11-error-L168 which seems to be related to the conditional switch here: https://github.com/clojure/clojure/blob/653b8465845a78ef7543e0a250078eea2d56b659/src/jvm/clojure/lang/Reflector.java#L45

I made a copy of the Java file and put it in `src-java`. It's compiled in `script/compile`.

- [x] Having  `java.nio.file.Files` in the reflection config, causes this error on macOS:

https://gist.github.com/borkdude/bc0d9941a9432d41a25665fc1ce9c760

Tl;dr: 

```
Undefined symbols for architecture x86_64:
  "_UTTypeCopyPreferredTagWithClass", referenced from:
      _Java_sun_nio_fs_UTIFileTypeDetector_probe0 in libnio.a(UTIFileTypeDetector.o)
  "_UTTypeCreatePreferredIdentifierForTag", referenced from:
      _Java_sun_nio_fs_UTIFileTypeDetector_probe0 in libnio.a(UTIFileTypeDetector.o)
  "_kUTTagClassFilenameExtension", referenced from:
      _Java_sun_nio_fs_UTIFileTypeDetector_probe0 in libnio.a(UTIFileTypeDetector.o)
  "_kUTTagClassMIMEType", referenced from:
      _Java_sun_nio_fs_UTIFileTypeDetector_probe0 in libnio.a(UTIFileTypeDetector.o)
```

Why is it trying to access sun.nio? 

Note that the same reflection.config works with a trivial Java program:

```
public class Program {
    public static void main(String [] args) {
        Class f = null;
        try { f = Class.forName("java.nio.file.Files");
        } catch (Exception e) {
        };
        System.out.println(f.toString());
        Class s = f.getSuperclass();
        System.out.println(s.getName());
    }
}
```

```
$ native-image --no-server -H:ReflectionConfigurationFiles=reflection.json --initialize-at-build-time Program
```

Fixed with

- [x] https://github.com/oracle/graal/issues/2015#issuecomment-575268745

Made an issue here: https://github.com/oracle/graal/issues/2015

- [x] Out of memory error for linux on CI:

https://circleci.com/gh/borkdude/babashka/3364

**Comment by borkdude** [comment]
Notes for Windows:

```
5:38 AM
sogaiu i had some success: https://pastebin.com/j91EAt1Z
i used your graalvm20-jdk11 branch (not the windows ones) plus a compile.bat file i had from before.
5:43 AM
sogaiu the compile.bat was: https://pastebin.com/dntwUBwR
5:44 AM
going to try your windows-graalvm20-jdk11 branch now
5:58 AM
sogaiu seems to work fine!  https://pastebin.com/GXRMheQs
5:59 AM
for both cases i did the build in: "x64 Native Tools Command Prompt for VS 2017"
```

**Comment by vincentjames501** [comment]
This would be great. Would love to see continued work on support for [hato](https://github.com/gnarroway/hato) as described in #220. I'd like to build a babashka websocket cli client using it ;) .

Is there a reason the dynamic requires cause issues with babashka? At first glance it appears to be working?

**Comment by borkdude** [comment]
@vincentjames501 I don't think hate will be part of babashka. It needs a whole lot of RAM to compile and adds about 8MB to the binary. I'd rather see someone implement it as a pod: https://github.com/borkdude/babashka/blob/master/doc/pods.md.
